### PR TITLE
Add method to obtain service account grant in GrantManager

### PR DESCRIPTION
**Pull request by rbarroetavena**
KEYCLOAK-4767
New method in order to get a grant for a service account using client
credentials. Request will use 'client_credentials' as the grant type.

Thanks

### CODE REVIEW

**Comment by abstractj** [comment]
@rbarroetavena could you please add functional tests?

**Comment by rbarroetavena** [comment]
@abstractj test added. thanks

File: test/integration/grant-manager-spec.js [hunk 1]
```diff
@@ -117,6 +117,16 @@ test('GrantManager in confidential mode should be able to validate an invalid to
     .then(t.end);
 });
```
Comment (abstractj) [inline test/integration/grant-manager-spec.js]: @rbarroetavena I'm sorry, but I don't see where you're testing `obtainFromClientCredentials` here.

File: test/integration/grant-manager-spec.js [hunk 2]
```diff
@@ -117,6 +117,16 @@ test('GrantManager in confidential mode should be able to validate an invalid to
...
 
+test('GrantManager in confidential mode should be able to obtain a service account grant', (t) => {
```
Comment (rbarroetavena) [inline test/integration/grant-manager-spec.js]: @abstractj don't quite get your point. the test uses obtainFromClientCredentials in order to grab a grant and then validates it. i didn't add the test you point in your comment. thanks

File: test/integration/grant-manager-spec.js [hunk 3]
```diff
@@ -117,6 +117,16 @@ test('GrantManager in confidential mode should be able to validate an invalid to
...
+  manager.obtainFromClientCredentials()
```
Comment (rbarroetavena) [inline test/integration/grant-manager-spec.js]: @abstractj this is where the new method gets called. thanks

Comment (rbarroetavena) [inline test/integration/grant-manager-spec.js]: hi @abstractj any comments/updates? thanks

Comment (abstractj) [inline test/integration/grant-manager-spec.js]: @rbarroetavena sorry about not getting back to you earlier. This is going to be reviewed next week.

**Comment by abstractj** [comment]
@rbarroetavena landed here https://github.com/keycloak/keycloak-nodejs-auth-utils/commit/9df010be0793b7738e456607234f752893f2ec86. Thank you!

**Pull request closed without merging**